style_editor.html

```diff
@@ -10018,16 +10077,117 @@
-          <input
-            type=button id=SLOW_BUTTON name=inhilt value="Slow"
-            title="Provides &quot;snapshots&quot; of rapidly moving things to see what they look like without fast movement. *Note* Graflex hilt model probably won't be able to render with this on."
-            onclick="ClickSlow()" >
+          <button 
+            id="SETTINGS_BUTTON" class="icon-button" title="Settings"
+            onclick="toggleSettingsPanel()">
+            <i class="fa-solid fa-cog"></i> Settings</button>
       </span>
 
+        <div id="settings_panel" class="settings-panel"
+          title="Turning on checkboxes saves your settings for future visits to the Style Editor page.">
+
+          <button
+           title="Close Settings Panel."
+            class="close-settings" onclick="toggleSettingsPanel()">X</button>
+
+            <h2 class="settings-header-label">Settings</h2>
+
+          <div class="settings-section">
+            <h4 class="settings-section-label" title="These settings apply to the blade preview above.">Blade Preview:</h4>
+
+            <label title="Simulates In-Hilt LED blades, (no addressable pixel effects).">
+              <input type="checkbox" id="INHILT_BUTTON" name="inhilt"
+		data-state="inhiltState" onclick="handleClick(this)">
+              <span>Inhilt LED</span>
+            </label>
+
+            <label title="Provides &quot;snapshots&quot; of rapidly moving things to see what they look like without fast movement. *Note* Graflex hilt model probably won't be able to render with this on.">
+              <input type="checkbox" id="SLOW_BUTTON" name="slow"
+		data-state="slowState" onclick="handleClick(this)">
+              <span>Slow</span>
+            </label>
+          </div>
+        </div>
+<script>
+
+function toggleSettingsPanel() {
+  var settingsButton = FIND("SETTINGS_BUTTON");
+  var settingsPanel = FIND("settings_panel");
+
+  settingsPanel.classList.toggle("show");
+
+  // Mouseleave event listener
+  if (settingsPanel.classList.contains("show")) {
+    var timeoutId = null;
+    settingsPanel.addEventListener("mouseleave", function(e) {
+      timeoutId = setTimeout(function() {
+        settingsPanel.classList.remove("show");
+      }, 1000);
+    });
+    settingsPanel.addEventListener("mouseenter", function(e) {
+      if (timeoutId) {
+        clearTimeout(timeoutId);
+        timeoutId = null;
+      }
+    });
+  }
+}
+
+// Call the onPageLoad function when the page is loaded
+window.addEventListener('DOMContentLoaded', onPageLoad);
+
+function onPageLoad() {
+  inhiltState.onload();
+  slowState.onload();
+}
+
+/* Settings buttons saved as local storage */
+function getSavedState(buttonState, defaultValue) {
+  var value = localStorage.getItem(buttonState);
+  console.log("Retrieved SavedState for " + buttonState + ": " + value);
+
+  if (value === null) {
+    return defaultValue;
+  }
+  // Convert the string to boolean unless it's null or "false"
```

Comment:
It can't be null when we get here...

Reply:
old comment from before the above line. deleted.